crates/proto/src/serialize/binary/encoder.rs

```diff
@@ -241,19 +241,18 @@ impl<'a> BinEncoder<'a> {
         assert!(end <= (u16::max_value() as usize));
         assert!(start <= end);
         if self.offset < 0x3FFF_usize {
-            self.name_pointers.push((start, end)); // the next char will be at the len() location
+            self.name_pointers.push((start, self.slice_of(start, end).to_vec())); // the next char will be at the len() location
```

Comment:
if `self::slice_of` returned a `&'a [u8]`, which it should be, I think we can just store the slice directly.

Reply:
Please see the above opened comment about data ownership here.